pkg/sypgp/sypgp.go

```diff
@@ -98,61 +105,67 @@ func PublicPath() string {
 	return filepath.Join(DirPath(), "pgp-public")
 }
 
-// PathsCheck creates the sypgp home folder, secret and public keyring files
-func PathsCheck() error {
-	// create the sypgp base directory
-	if err := os.MkdirAll(DirPath(), 0700); err != nil {
+func ensureDirPrivate(dn string) error {
+	mode := os.FileMode(0700)
+
+	if err := os.MkdirAll(dn, mode); err != nil {
 		return err
 	}
 
-	dirinfo, err := os.Stat(DirPath())
+	dirinfo, err := os.Stat(dn)
 	if err != nil {
 		return err
 	}
-	if dirinfo.Mode() != os.ModeDir|0700 {
-		sylog.Warningf("directory mode (%v) on %v needs to be 0700, fixing that...", dirinfo.Mode(), DirPath())
-		if err = os.Chmod(DirPath(), 0700); err != nil {
+
+	if currentMode := dirinfo.Mode(); currentMode != os.ModeDir|mode {
+		sylog.Warningf("directory mode (%o) on %s needs to be %o, fixing that...", currentMode & ^os.ModeDir, dn, mode)
+		if err = os.Chmod(dn, mode); err != nil {
 			return err
 		}
 	}
 
-	// create or open the secret OpenPGP key cache file
-	fs, err := os.OpenFile(SecretPath(), os.O_RDWR|os.O_CREATE, 0600)
+	return nil
+}
+
+func ensureFilePrivate(fn string) error {
+	mode := os.FileMode(0600)
+
+	fs, err := os.OpenFile(fn, os.O_RDWR|os.O_CREATE, mode)
```

Comment:
same here

Reply:
this is resolved, dismissing in order to be able to move.